nextjs-13/src/repositories/newsletter.repository.ts

```diff
@@ -6,4 +6,13 @@ const createNewsletter = async (
   newsletterAttributes: Prisma.NewsletterCreateInput
 ) => dbClient.newsletter.create({ data: newsletterAttributes });
 
-export { createNewsletter };
+const queryNewsletterByUserId = async (id: string) => {
```

Comment:
I think you can change this function name to `findNewsletterByUserId` instead of `queryNewsletterByUserId`.  This makes it more like natural language and easier to understand.

[Reference](https://stackoverflow.com/questions/2141818/method-names-for-getting-data)

Reply:
Ya although `query` does not sound as easy as `find`, I think `find` is more suitable for cases when we only need to get only one record, and `query` is something that is more suitable for getting many records at once, like, when we want to get some data(joins table bla bla) by doing some raw sql, we usually says "query sql".

And maybe the reason I think so is because I'm so used to Ruby on Rails, and RoR use [`find` to only get one record](https://apidock.com/rails/ActiveRecord/Base/find/class), and [`where` to get many records](https://apidock.com/rails/ActiveRecord/QueryMethods/where) 😄. I'm an advocate for using 2 different verbs for 2 different kinds of getting data(get one and many), as it helps us need less brain power.

For example, in Rails way, we just need to read the first verb of a function name (`find` or `query`) to know if it queries 1 record or many. If we use the same `find` for both kinds of query, like `findNewsletter` and `findNewsletters`, we have to read until the last `s` word to know that if this is getting one or many 🤔. `findNewsletter` and `queryNewsletters` are faster to differentiate to me.

So this is just a personal preference thing, I think 👀

Reply:
Oh okay, Thanks for sharing 😁. 

But for this function, you are using `findUnique` in Prisma, which will return only one record, so I think `find` will be good here as you mention the use case to use `find` above.

Reply:
Ya, but the thing this function return is the newsletters of the user, which is plural 👀 It may cause confusion a bit here, as in later PRs, I refactored it to be:

```typescript
dbClient.newsletter.findMany({
    where: { userId: userId }
})
```

Reply:
Noted